### ISSUE TITLE
Some text from plan is missing in .dox download

### ISSUE THREAD

**Issue by bofstein** [CDLUC3/dmptool#647]
**Describe the bug**
Some text of plan is missing in .docx download of DMP

**To Reproduce**
1. Log in as Super Admin to DMP Tool
2. Go to this DMP (unsure if happening to others): https://dmptool.org/plans/122390/download
3. Download plan as PDF, CSV, Text file, or HTML 
4. Download plan as Word doc
5. Observe that the following blocks of text are present in the other files (and in the plan narrative in tool) but not in the Word doc: 

"Biochemical Assay Data: Results from various biochemical assays assessing the binding and cleaving activities of NLRP3 and other pyrin-domain proteins on oxidized DNA. 
Cell Signaling Data: Measurements from cell-based assays evaluating the activation of NLRP3 inflammasome and downstream signaling pathways under different stress conditions."

"Public Repositories: Published data will be deposited in publicly accessible repositories such as the Electron Microscopy Data Bank (EMDB) and the Protein Data Bank (PDB) for long-term access and preservation.
Constructs and Reagents:"

"Direct Requests: Researchers can email _redacted name for privacy_ to request reagents and materials needed to duplicate the data, facilitating reproducibility and collaboration."

"Protein Data Bank (PDB): Atomic models derived from cryo-EM studies will be deposited in the PDB. The PDB is a well-established repository for 3D structural data of biological macromolecules, making these models widely accessible to the scientific community."

**Expected behavior**

**Screenshots**
This is the missing text, which does appear in the plan in the tool
<img width="763" alt="Screenshot 2024-07-11 at 5 31 40 PM (002)" src="https://github.com/user-attachments/assets/171bf728-9666-4ab0-bbde-57680986b5b9">

**Additional context**
Unsure if this is happening to others, was reported by an admin of the org the plan is from.

**Comment by bofstein** [comment]
I made a new DMP on Stage and copied in the text of the bugged plan into this one.  For the first question, I copied the formatting, adding bold wherever the original plan had it.  For the other questions I just pasted in the plain text. 

https://dmptool-stg.cdlib.org/plans/89637/download

In this new plan, the first question (where I repeated formatting) is still missing the same information, while the other missing blocks appear to be in the .docx file now.  So something to do with formatting perhaps?

**Comment by bofstein** [comment]
@briri and @jupiter007 I added you as editors to my copy of the plan on Stage - that's just for when we get to this bug so you have it to work from, not that anything needs to be done right away before we schedule it.

**Comment by briri** [comment]
Thanks for doing that analysis @bofstein. I suspect that the DOCX library we use is having trouble figuring out how to render the formatted portion of the text and is just skipping it rather than generating an error.

**Comment by bofstein** [comment]
Start with investigation to determine the workload to fix it

**Comment by bofstein** [comment]
@jupiter007 to take a look this sprint if time (e.g. if tasks finish or waiting on PR or want a break from other tasks)

**Comment by jupiter007** [comment]
I did a little bit of research on this. I copied a couple of the paragraphs over from https://dmptool.org/plans/122390/edit?phase_id=1650  that contained missing data in the "`docx`" formatted file. It turns out that some combination of `<strong>` tags with `<br/> `tags within the same `<p>` tag appears to be causing an issue in the resulting DOCX file. When a `<p>` tag contains both the `<strong>` tags and `<br/>` tags, it causes the text after the `<br/> `tag to not be included in DOCX file.

For example, I copied over the following text in red to my own plan:
<img width="853" alt="image" src="https://github.com/user-attachments/assets/ca0b12ff-b978-462d-8124-c99e9cf8005b">

I then downloaded it in "**docx**" format and confirmed that the following text was missing: **"Biochemical Assay Data: Results from various biochemical assays assessing the binding and cleaving activities of NLRP3 and other pyrin-domain proteins on oxidized DNA.
Cell Signaling Data: Measurements from cell-based assays evaluating the activation of NLRP3 inflammasome and downstream signaling pathways under different stress conditions."**

I then removed the "**bold**" on the "**Cryo-Electron Microscopy (Cryo-EM) Data:**" text (i.e., I removed the `<strong>` tag from that paragraph), and then downloaded the "`docx`" file again, and this time the formerly missing text displayed.

You can see where the `<strong>` and `<br>` tags are in the **Rich Text Editor** if you look in the "**Elements**" tab of the browser's **Developer Tool**, and search for the the relevant text.

I did the same test on a paragraph that didn't have a problem initially, which is:
<img width="853" alt="image" src="https://github.com/user-attachments/assets/77a71955-b5c4-4923-a71f-984c47e90414">

I copied that text over, and bolded the "**Image Processing Software:**" text, which caused the text after the `<br>` tag to be excluded from final doc: **"Data Analysis Scripts: Custom code written in Python, MATLAB, or other programming languages for analyzing experimental data."**

**Comment by jupiter007** [comment]
We could possibly solve this problem by replacing `<br>` tags with `<p>` tags for DOCX files.

**Comment by bofstein** [comment]
Assigning to @jupiter007 for once she's back from UC Tech to finish and plan release this sprint

**Comment by jupiter007** [comment]
I fixed the issue by adding this substitution: `html.gsub(/(<p>.*?<strong>.*?<\/strong>[^<]+)(<br>)(.*?<\/p>)/, '\1</p><p>\3')`.

Testing:

1. I copied over the problematic content referenced in the ticket, into my plan.

"Biochemical Assay Data: Results from various biochemical assays assessing the binding and cleaving activities of NLRP3 and other pyrin-domain proteins on oxidized DNA.
Cell Signaling Data: Measurements from cell-based assays evaluating the activation of NLRP3 inflammasome and downstream signaling pathways under different stress conditions."
"Public Repositories: Published data will be deposited in publicly accessible repositories such as the Electron Microscopy Data Bank (EMDB) and the Protein Data Bank (PDB) for long-term access and preservation.
Constructs and Reagents:"

"Direct Requests: Researchers can email redacted name for privacy to request reagents and materials needed to duplicate the data, facilitating reproducibility and collaboration."

2. I then downloaded a .docx file WITHOUT my regex fix. You can see where the missing content should be in the outlined red boxes:
![Image](https://github.com/user-attachments/assets/749ee90d-08fa-488a-8ada-64163a86f9c3)

3. Then I added my regex fix, and downloaded the .docx file again. You can see the missing content is now present, and the formatting looks the same as in the originally entered text in the plan:
 ![Image](https://github.com/user-attachments/assets/a4b33c15-2557-4c30-85f3-ddddc9273f87)

**Comment by jupiter007** [comment]
Changes were released to production in v5.27.
